bee-storage/bee-storage-rocksdb/src/persistable.rs

```diff
@@ -122,92 +122,147 @@ where
 }
 
 impl Persistable<Storage> for TransactionMetadata {
-    fn encode_persistable<Storage>(&self, buffer: &mut Vec<u8>) {
+    fn write_to(&self, buffer: &mut Vec<u8>) {
         // encode struct in order
         // 1- encode flags
-        self.flags().bits().encode_persistable::<Storage>(buffer);
+        self.flags().bits().write_to(buffer);
         // 2- encode milestone_index
-        self.milestone_index().encode_persistable::<Storage>(buffer);
+        self.milestone_index().write_to(buffer);
         // 3- encode arrival_timestamp
-        self.arrival_timestamp().encode_persistable::<Storage>(buffer);
+        self.arrival_timestamp().write_to(buffer);
         // 4- encode solidification_timestamp
-        self.solidification_timestamp().encode_persistable::<Storage>(buffer);
+        self.solidification_timestamp().write_to(buffer);
         // 5- encode confirmation_timestamp
-        self.confirmation_timestamp().encode_persistable::<Storage>(buffer);
+        self.confirmation_timestamp().write_to(buffer);
+        // 6- encode cone_index
+        if let Some(cone_index) = self.cone_index() {
+            cone_index.write_to(buffer)
+        }
+        // 7- encode otrsi
+        if let Some(otrsi) = self.otrsi() {
+            otrsi.write_to(buffer)
+        }
+        // 8- encode ytrsi
+        if let Some(ytrsi) = self.ytrsi() {
+            ytrsi.write_to(buffer)
+        }
```

Comment:
Why did we write a `impl Persistable<Storage> for Option<MilestoneIndex>` if we're not using it here ? Shouldn't we still write something when None ?

Reply:
oh yeah this is my mistake regarding the `Option<Milestone>` which I implemented but didn't make use of it
gonna fix it now.

Reply:
Also Option<T> indicates both Some(t) and None.